src/index.js

```diff
@@ -31,8 +31,10 @@ const isolateSink =
   (sink, scope) =>
     sink.map(
       vtree => {
-        const c = `${vtree.sel} cycle-scope-${scope}`.trim()
-        vtree.sel = c
+        if (vtree.sel.indexOf(`cycle-scope-${scope}`) === -1) {
+          const c = `${vtree.sel}.cycle-scope-${scope}`.trim()
```

Comment:
You might not need to trim() here since `sel` is not a className, and classNames are the ones that have spaces around.